# Argument(ation) mining

## Table of Contents(maybe remove)

- [Argument Components](#arg_comp)
    * [Load Data](#load_data)
    * [Components distribution](#comp_dist)
    * [Transformer-based Argument Mining for Healthcare Applications](#ecai2020)
    * [Argument components classification results](#comp_clf)
        - Results: pre-problem identification
        - Results: post-problem identification
* [Translation and projections](#translation_projection)
    - Translations
    - Projections
- [Zero-shot](#zero-shot)
    - Train (ES) + test(ES)
    - BETO
    - Data augumentation results => train(EN + ES), test(ES), test(EN)
    - 
    
* [Relation Classification](#rel_clf)
    - Results

<a class='anchor' id='arg_comp'></a>
# Argument Components 

In [1]:
import pandas as pd
import glob
import os

<a id='load_data'></a>
### Load data 

Load AbstRCT data that contains .txt and .ann files for each category of disease 

In [2]:
neoplasm_trn = '../abstrct-master/AbstRCT_corpus/data/train/neoplasm_train/'
neoplasm_tst = '../abstrct-master/AbstRCT_corpus/data/dev/neoplasm_dev/'
neoplasm_dv = '../abstrct-master/AbstRCT_corpus/data/test/neoplasm_test/'
mixed = '../abstrct-master/AbstRCT_corpus/data/test/mixed_test/'
glaucoma = '../abstrct-master/AbstRCT_corpus/data/test/glaucoma_test/'


def get_annotated_data(url):
    files = glob.glob(url+"/*.ann")

    
    df = pd.DataFrame()
    for f in files:

        df_temp = pd.read_csv(f, sep='\t', header = None)
        df = df.append(df_temp)
        df['url'] = f[50:-4]
    df.dropna(inplace=True)
    df.reset_index(inplace=True)
    return df

# Neoplasm
neoplasm_train = get_annotated_data(neoplasm_trn)
neoplasm_test = get_annotated_data(neoplasm_tst)
neoplasm_dev = get_annotated_data(neoplasm_dv)

# Glaucoma
glaucoma_test = get_annotated_data(glaucoma)

# Mixed
mixed_test = get_annotated_data(mixed)

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9sqv80000gn/T/ipykernel_74472/848597918.py:16: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df_temp)
/var/folders/lj/zwbh4h2d2v147nv8rsp9

In [3]:
neoplasm_train.head()

index   0                  1  \
0      0  T1    Premise 761 825   
1      1  T2    Premise 826 932   
2      2  T3   Premise 933 1107   
3      3  T4  Premise 1108 1248   
4      4  T5  Premise 1249 1437   

                                                   2                url  
0  The overall remission rate was 87% with 31% co...  sm_train/15920748  
1  The median survival of all 406 eligible patien...  sm_train/15920748  
2  The overall remission rate, the rate of comple...  sm_train/15920748  
3  In limited disease the estimated percentages o...  sm_train/15920748  
4  Patients with extensive disease survived signi...  sm_train/15920748

In [4]:
glaucoma_test.head()

index   0                  1  \
0      0  T1    Premise 683 812   
1      1  T2    Premise 813 982   
2      2  T3   Premise 983 1130   
3      3  T4  Premise 1131 1292   
4      4  T5  Premise 1293 1378   

                                                   2              url  
0  Percentage of IOP reduction or the magnitude o...  a_test/16899528  
1  In the visual field, the estimated rate of cha...  a_test/16899528  
2  The estimated rate of change in MD showed no s...  a_test/16899528  
3  No changes in the optic nerve head topography ...  a_test/16899528  
4  There were no patients who dropped out due to ...  a_test/16899528

In [5]:
# Neoplasm

# neoplasm_train.info() #  2267
# neoplasm_test.info() # 326
# neoplasm_dev.info() # 686

# Glaucoma
# glaucoma_test.info() # 594

# Mixed
mixed_test.info()  # 600

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 600 entries, 0 to 599
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   index   600 non-null    int64 
 1   0       600 non-null    object
 2   1       600 non-null    object
 3   2       600 non-null    object
 4   url     600 non-null    object
dtypes: int64(1), object(4)
memory usage: 23.6+ KB


In [6]:
neoplasm_dev.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 686 entries, 0 to 685
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   index   686 non-null    int64 
 1   0       686 non-null    object
 2   1       686 non-null    object
 3   2       686 non-null    object
 4   url     686 non-null    object
dtypes: int64(1), object(4)
memory usage: 26.9+ KB


**Add column of relation component type**

In [7]:
def relation_components(df):
    components = list(df[1])
    comp_list = []
    for c in components:
        comp_list.append(c.split()[0])
    df['relation_components'] = comp_list

    return df

neoplasm_train = relation_components(neoplasm_train)
neoplasm_test = relation_components(neoplasm_test)
neoplasm_dev = relation_components(neoplasm_dev)

glaucoma_test = relation_components(glaucoma_test)

mixed_test = relation_components(mixed_test)

In [8]:
mixed_test.head()

index   0                  1  \
0      0  T1   Premise 763 1114   
1      1  T2  Premise 1188 1295   
2      2  T3  Premise 1296 1363   
3      3  T4    Claim 1364 1497   
4      0  T1  Premise 1287 1583   

                                                   2           url  \
0  In sildenafil versus placebo arms, week-12 6MW...  est/28371482   
1  Changes in WHO functional class and Borg dyspn...  est/28371482   
2  Headache, diarrhoea, and flushing were more co...  est/28371482   
3  Sildenafil, in addition to stable (≥3 months) ...  est/28371482   
4  Responses (> or = 50% improvement) were seen i...  est/28371482   

  relation_components  
0             Premise  
1             Premise  
2             Premise  
3               Claim  
4             Premise

<a id='comp_dist'></a>
### Component distributions

In [9]:
print("Neoplasm train:\n\n", neoplasm_train['relation_components'].value_counts(), '\n')
print("Neoplasm test:\n\n" , neoplasm_test['relation_components'].value_counts(), '\n')
print("Neoplasm dev: \n\n", neoplasm_dev['relation_components'].value_counts(), '\n')
# print('\n')
print("Glaucoma test: \n\n",glaucoma_test['relation_components'].value_counts(), '\n')
# print('\n')
print('Mixed test: \n\n', mixed_test['relation_components'].value_counts(), '\n')

Neoplasm train:

 Premise       1537
Claim          666
MajorClaim      64
Name: relation_components, dtype: int64 

Neoplasm test:

 Premise       218
Claim          99
MajorClaim      9
Name: relation_components, dtype: int64 

Neoplasm dev: 

 Premise       438
Claim         228
MajorClaim     20
Name: relation_components, dtype: int64 

Glaucoma test: 

 Premise       404
Claim         183
MajorClaim      7
Name: relation_components, dtype: int64 

Mixed test: 

 Premise       388
Claim         182
MajorClaim     30
Name: relation_components, dtype: int64 



In [10]:
print('Mean number of symbols per line - NEOPLASM: \n')
print("\t train set: ", int(neoplasm_train[2].apply(lambda x: len(x)).mean()))
print("\t test set: ", int(neoplasm_test[2].apply(lambda x: len(x)).mean()))
print("\t dev set: ", int(neoplasm_dev[2].apply(lambda x: len(x)).mean()))
print('\n')
print('Mean number of symbols per line - GLAUCOMA: ', int(glaucoma_test[2].apply(lambda x: len(x)).mean()))
print('\n')
print('Mean number of symbols per line - MIXED: ', int(mixed_test[2].apply(lambda x: len(x)).mean()))


Mean number of symbols per line - NEOPLASM: 

	 train set:  138
	 test set:  143
	 dev set:  135


Mean number of symbols per line - GLAUCOMA:  138


Mean number of symbols per line - MIXED:  145


In [11]:
print('Mean number of words per line - NEOPLASM: \n')
print("\t train set: ", int(neoplasm_train[2].apply(lambda x: len(x.split())).mean()))
print("\t test set: ", int(neoplasm_test[2].apply(lambda x: len(x.split())).mean()))
print("\t dev set: ", int(neoplasm_dev[2].apply(lambda x: len(x.split())).mean()))
print('\n')
print('Mean number of words per line - GLAUCOMA: ', int(glaucoma_test[2].apply(lambda x: len(x.split())).mean()))
print('\n')
print('Mean number of words per line - MIXED: ', int(mixed_test[2].apply(lambda x: len(x.split())).mean()))

Mean number of words per line - NEOPLASM: 

	 train set:  21
	 test set:  22
	 dev set:  20


Mean number of words per line - GLAUCOMA:  21


Mean number of words per line - MIXED:  22


**[ Biran and Rambow (2011)] found that premises are longer on the average than other sentences.** Yes

In [12]:
print('Neoplasm train: \n')
print('Mean length of MajorClaim: ',neoplasm_train[neoplasm_train['relation_components'] == 'MajorClaim'][2].apply(lambda x: len(x.split())).mean())
print('Mean length of Claim: ',neoplasm_train[neoplasm_train['relation_components'] == 'Claim'][2].apply(lambda x: len(x.split())).mean())
print('Mean length of Premise: ',neoplasm_train[neoplasm_train['relation_components'] == 'Premise'][2].apply(lambda x: len(x.split())).mean())

Neoplasm train: 

Mean length of MajorClaim:  16.484375
Mean length of Claim:  19.064564564564563
Mean length of Premise:  22.657124268054652


In [13]:
print('Glaucoma test: \n')
print('Mean length of MajorClaim: ',glaucoma_test[glaucoma_test['relation_components'] == 'MajorClaim'][2].apply(lambda x: len(x.split())).mean())
print('Mean length of Claim: ',glaucoma_test[glaucoma_test['relation_components'] == 'Claim'][2].apply(lambda x: len(x.split())).mean())
print('Mean length of Premise: ',glaucoma_test[glaucoma_test['relation_components'] == 'Premise'][2].apply(lambda x: len(x.split())).mean())

Glaucoma test: 

Mean length of MajorClaim:  16.142857142857142
Mean length of Claim:  16.950819672131146
Mean length of Premise:  23.925742574257427


In [14]:
print('Mixed test: \n')
print('Mean length of MajorClaim: ',mixed_test[mixed_test['relation_components'] == 'MajorClaim'][2].apply(lambda x: len(x.split())).mean())
print('Mean length of Claim: ',mixed_test[mixed_test['relation_components'] == 'Claim'][2].apply(lambda x: len(x.split())).mean())
print('Mean length of Premise: ',mixed_test[mixed_test['relation_components'] == 'Premise'][2].apply(lambda x: len(x.split())).mean())

Mixed test: 

Mean length of MajorClaim:  17.633333333333333
Mean length of Claim:  17.78021978021978
Mean length of Premise:  24.75


In [24]:
from nltk.corpus import stopwords

In [53]:
claim_word_freq = {}
claims_in_train = neoplasm_train[neoplasm_train['relation_components'] == 'Claim']
# claims_in_train.head()
for i in claims_in_train[2]:
    line = i.split()
    filtered_line = [word for word in line if word.lower() not in stopwords.words('english')]
    for l in filtered_line:
        if l not in claim_word_freq:
            claim_word_freq[l] = 1
        else:
            claim_word_freq[l] += 1
    

In [54]:
{k : j for k,j in sorted(claim_word_freq.items(), key=lambda item: item[1], reverse=True)}

{'patients': 176,
 'treatment': 128,
 'cancer': 107,
 'survival': 79,
 'advanced': 62,
 'quality': 62,
 'may': 54,
 'breast': 51,
 'compared': 50,
 'significant': 44,
 'life': 43,
 'better': 40,
 'significantly': 40,
 'QOL': 40,
 'therapy': 38,
 'associated': 38,
 'effective': 36,
 'improve': 36,
 'results': 32,
 'study': 32,
 'overall': 31,
 'improved': 31,
 'chemotherapy': 29,
 'clinical': 29,
 'similar': 29,
 'cancer.': 28,
 'symptoms': 28,
 'response': 27,
 'effect': 26,
 'higher': 25,
 'patients.': 25,
 'women': 25,
 'combination': 25,
 'patient': 25,
 'intervention': 23,
 'improvement': 23,
 'suggest': 21,
 'well': 21,
 'plus': 20,
 'improves': 20,
 'efficacy': 20,
 'effects': 20,
 'months': 19,
 'physical': 19,
 'treated': 19,
 'impact': 19,
 'However,': 18,
 'receiving': 18,
 'benefit': 18,
 'regimen': 18,
 'clinically': 17,
 'adjuvant': 17,
 'metastatic': 17,
 'cancer,': 16,
 'differences': 16,
 'care': 16,
 'terms': 16,
 'shown': 16,
 'advantage': 16,
 'less': 16,
 'time': 16

In [51]:
premise_word_freq = {}
premise_in_train = neoplasm_train[neoplasm_train['relation_components'] == 'Premise'].replace('[^a-zA-Z]', '')
# claims_in_train.head()
for i in premise_in_train[2]:
    line = i.split()
    filtered_line = [word for word in line if word.lower() not in stopwords.words('english')]
    for l in filtered_line:
        if l not in premise_word_freq:
            premise_word_freq[l] = 1
        else:
            premise_word_freq[l] += 1

In [52]:
{k : j for k,j in sorted(premise_word_freq.items(), key=lambda item: item[1], reverse=True)}

{'=': 588,
 'patients': 365,
 'group': 342,
 '(P': 295,
 'P': 255,
 'months': 221,
 'significant': 217,
 'survival': 214,
 'significantly': 200,
 '<': 184,
 'treatment': 152,
 'difference': 145,
 '95%': 137,
 'compared': 129,
 'scores': 126,
 'arm': 121,
 '(p': 110,
 'CI,': 110,
 'versus': 109,
 '(95%': 109,
 'median': 107,
 'differences': 104,
 'CI': 102,
 'overall': 95,
 'groups': 95,
 'v': 92,
 'quality': 91,
 'time': 90,
 'higher': 88,
 'life': 87,
 '3': 84,
 'improved': 83,
 'statistically': 82,
 'control': 80,
 'response': 76,
 'vs': 71,
 'pain': 70,
 'QOL': 69,
 'reported': 67,
 'showed': 67,
 'months,': 66,
 'observed': 65,
 'baseline': 65,
 'rate': 64,
 '2': 64,
 'mean': 64,
 'group.': 61,
 'better': 61,
 'receiving': 58,
 'placebo': 58,
 'group,': 57,
 'intervention': 57,
 'rates': 57,
 '6': 56,
 'treated': 55,
 'vs.': 55,
 '1': 54,
 'Patients': 53,
 'lower': 53,
 'two': 52,
 'weeks': 52,
 'improvement': 51,
 'physical': 51,
 'confidence': 50,
 'ratio': 50,
 'increased': 49,


<a id='ecai2020'></a>
## <a href='https://ecai2020.eu/papers/1470_paper.pdf'>Transformer-based Argument Mining for Healthcare Applications</a>


Check and compare the data compliance with data from previous section

### Neoplasm

<!-- Glaucoma test:  {'Premise': 404, 'Claim': 183, 'MajorClaim': 7} -->


<!-- Mixed test:  {'Premise': 388, 'Claim': 182, 'MajorClaim': 30} -->

In [49]:
train_conll = pd.read_csv('ecai2020-transformer_based_am/data/neoplasm/train.conll', sep='\t', header=None)
test_conll = pd.read_csv('ecai2020-transformer_based_am/data/neoplasm/test.conll', sep='\t', header=None)
dev_conll = pd.read_csv('ecai2020-transformer_based_am/data/neoplasm/dev.conll', sep='\t', header=None)
train_conll.head()

0          1  2  3        4
0  0     Facial  _  _  B-Claim
1  1  hirsutism  _  _  I-Claim
2  2         is  _  _  I-Claim
3  3        one  _  _  I-Claim
4  4         of  _  _  I-Claim

In [50]:
print("Train: \n", train_conll[4].value_counts())

Train: 
 O            61110
I-Premise    42438
I-Claim      14383
B-Premise     1535
B-Claim        729
Name: 4, dtype: int64


 > Supposed to be : **B-premise** = 1537, **B-claim** = 730 

In [51]:
print("Test: \n",test_conll[4].value_counts())

Test: 
 O            16739
I-Premise    11931
I-Claim       4677
B-Premise      438
B-Claim        248
Name: 4, dtype: int64


> Test => **Premise**: 218, **Claim**: 108 (99 + 9 (MajorClaim)) 

Should be compeletly different numbers according to the files in (.txt and .ann). 

***Looks like in .conll test is dev in .ann and vice versa.*** Yes!

In [18]:
print("Dev: \n",dev_conll[4].value_counts())

Dev: 
 O            9124
I-Premise    6655
I-Claim      2087
B-Premise     218
B-Claim       108
Name: 4, dtype: int64


(dev, test = test, dev)

>  Dev => **Premise**: 438, **Claim**: 248 (228 + 20(**MajorClaim**))

______________
Dev => Premise: 218, Claim: 108 <br>
Test => Premise: 438, Claim: 248
______________


### Glaucoma

In [52]:
gl_conll = pd.read_csv('ecai2020-transformer_based_am/data/glaucoma_test/test.conll', sep='\t', header=None)
gl_conll.head()

0      1  2  3  4
0  1    The  _  _  O
1  2    aim  _  _  O
2  3     of  _  _  O
3  4   this  _  _  O
4  5  study  _  _  O

In [53]:
print("Glaucoma: \n", gl_conll[4].value_counts())

Glaucoma: 
 O            15919
I-Premise    11483
I-Claim       3357
B-Premise      404
B-Claim        190
Name: 4, dtype: int64


> Glaucoma test:  **Premise** -> 404; **Claim**: 190 (183 + 7 (MajorClaim))

### Mixed

In [23]:
mx_conll = pd.read_csv('ecai2020-transformer_based_am/data/mixed_test/test.conll', sep='\t', header=None)
mx_conll.head()

0           1  2  3  4
0  0          We  _  _  O
1  1  previously  _  _  O
2  2    reported  _  _  O
3  3        that  _  _  O
4  4   treatment  _  _  O

In [24]:
print("Mixed: \n", mx_conll[4].value_counts())

Mixed: 
 O            16015
I-Premise    11767
I-Claim       3942
B-Premise      388
B-Claim        212
Name: 4, dtype: int64


> Mixed test:  **Premise**: 388, **Claim** : 212 (182 + 30 (MajorClaim))

### Find and fix the error in data

**Problem data due to the incorrect splitting in the data**

Tokens have been splitted incorrectly somewhere. For example:
    
 ***and nine symptoms ( .001 < P < . 01) , and the improvement***  - extra space in P < . 01 forces unnecessary split in the data which breaks IOB-schema logic

In [54]:
def check(url):
    '''
    Find the lines that cause the problem
    '''
    check_lines = []
    with open(url, 'r') as f:
        lines = f.readlines()
    for i, line in enumerate(lines[:-1]):
        if line != '\n':
            _, tkn, _, _, tag = line.split('\t')
            
        elif line == '\n' and lines[i+1] != '\n' and lines[i+1].split('\t')[4].startswith('I-'):
            check_lines.append(i+1)
    return check_lines

In [55]:
train_check = check('ecai2020-transformer_based_am/data/neoplasm/train.conll')
train_check

[]

#### Neoplasm 

<!-- 
# train_conll[train_conll[0] == 279][train_conll[1] == '4'] 
# train_conll[train_conll[0] == 359][train_conll[1] == '0001'] # 10129	359	0001	_	_	I-Premise
# train_conll[train_conll[0] == 400][train_conll[1] == '0002'] # 10170	400	0002	_	_	I-Premise
# train_conll[train_conll[0] == 213][train_conll[1] == '93'] # 11510	213	93	_	_	I-Premise
# train_conll[train_conll[0] == 242][train_conll[1] == '01'] # 15080	242	01	_	_	I-Premise
# train_conll[train_conll[0] == 271][train_conll[1] == '004'] # 33253	271	004	_	_	I-Premise
# train_conll[train_conll[0] == 234][train_conll[1] == 'US'] # 34268	234	US	_	_	I-Premise
# train_conll[train_conll[0] == 213][train_conll[1] == '048'] # 48940	213	048	_	_	I-Premise
# train_conll[train_conll[0] == 189][train_conll[1] == 'pain'] # 51615	189	pain	_	_	I-Premise
# train_conll[train_conll[0] == 210][train_conll[1] == '12'] # 51951	210	12	_	_	I-Premise
# train_conll[train_conll[0] == 232][train_conll[1] == '7'] # 51973	232	7	_	_	I-Premise
# train_conll[train_conll[0] == 205][train_conll[1] == 'better'] # 92762	205	better	_	_	I-Premise
 -->

In [189]:
def find_lines(arr, data):
    '''
    find lines that are causing problems to find out why
    prints number of the line in the file where the incorrect labels begin 
    returns 10 tokens before and after the problematic line
    '''
    chunk = []
    with open(data) as f:
            lines = f.readlines()
    for idx in arr:
        start = idx - 10
        end = idx + 10
        print(start)
        c = ''
        for i, line in enumerate(lines): 
           
            if i >= start and i <= end:
                if line != '\n':
                    c += line.split('\t')[1] + ' '
        chunk.append(c)
                    
    return chunk
        
lines_to_check = find_lines(check_lines, 'ecai2020-transformer_based_am/data/neoplasm/train.conll')

10434
10517
10560
11948
15662
34565
35624
50891
53688
54037
54061
96489


In [190]:
lines_to_check

['95 % CI = -9.6 % to -1 . 4 % ) per 4 weeks in the control group but ',
 '=.0002 , P =.0001 , and P = . 0001 , respectively , for ATP versus no ATP ) . ',
 '-0.2 % versus -2.4 % ; P = . 0002 ) ; functional scores ( +0.4 % versus -5.5 % ',
 '; 95 % CI , 0.61 to 0 . 93 ; P = .008 ) . Median survival was ',
 'and nine symptoms ( .001 < P < . 01 ) , and the improvement ( > 10 units on ',
 'responders indicated better physical well-being ( P = . 004 ) and mood ( P =.02 ) at month 3 ',
 'patients than in the control group , i.e . US $ 40782 and US $ 34465 , respectively ( P ',
 'survival rate , 31.8 % ; P = . 048 ) . Comparing survival for the two dose levels ',
 'and resolved . Mean ( s.d . ) pain scores on the day of discharge were 1·9 ( 3·1 ',
 ', 16.2 v 14.7 months ; P = . 12 ) . LV5FU2 plus oxaliplatin gave higher frequencies of ',
 'common toxicity criteria grade 3/4 neutropenia ( 41 . 7 % v 5.3 % of patients ) , grade 3/4 ',
 'QOL ( p < 0.05 ) ( i.e . better global health status , an

***Sequence where something is wrong => corresponding file where these lines are located***

'95 % CI = -9.6 % to -1 . 4 % ) per 4 weeks in the control group but ' => ***10675381.ann***   
'=.0002 , P =.0001 , and P = . 0001 , respectively , for ATP versus no ATP ) . ' => ***10675381.ann***   
'-0.2 % versus -2.4 % ; P = . 0002 ) ; functional scores ( +0.4 % versus -5.5 %'  => ***10675381.ann***   
'; 95 % CI , 0.61 to 0 . 93 ; P = .008 ) . Median survival was ' => ***15625369.ann***   
'and nine symptoms ( .001 < ***P < . 01***) , and the improvement ( > 10 units on ' => ***10561201.ann***   
responders indicated better physical well-being **( P = . 004 )** and mood ( P =.02 ) at month 3 => ***10561203.ann***   
patients than in the control group , i.e . US $ 40782 and US $ 34465 , respectively ( P => ***9531327.ann***   
survival rate , 31.8 % ; P = **. 048 )** . Comparing survival for the two dose levels => ***10653877.ann***   
and resolved . Mean ( s.d . ) pain scores on the day of discharge were 1·9 ( 3·1 => ***23254324.ann***   
, 16.2 v 14.7 months ; P = . 12 ) . LV5FU2 plus oxaliplatin gave higher frequencies of => ***10944126.ann***   
common toxicity criteria grade 3/4 neutropenia ( 41 . 7 % v 5.3 % of patients ) , grade 3/4 => ***10944126.ann***  
QOL ( p < 0.05 ) ( i.e . better global health status , and physical and emotional functioning => ***23866850.ann***  

In [196]:
# print(check_lines)

[10444, 10527, 10570, 11958, 15672, 34575, 35634, 50901, 53698, 54047, 54071, 96499]


***Dev and Test***

In [29]:
test = 'ecai2020-transformer_based_am/data/neoplasm/test.conll'
dev = 'ecai2020-transformer_based_am/data/neoplasm/dev.conll'

In [30]:
test_check = check(test)
dev_check = check(dev)

In [31]:
test_check

[1807, 1816, 1834, 1843, 1856, 1865, 33747]

In [214]:
dev_check

[]

In [216]:
tst_find_lines = find_wrong_lines(test_check, test)
tst_find_lines

1797
1806
1824
1833
1846
1855
33737


['both arms , with 12 % ( postop . RCT ) and 12 % ( pre-op . RCT ) ',
 'RCT ) and 12 % ( pre-op . RCT ) of patients , respectively , suffering from anastomotic leakage ',
 'from anastomotic leakage , 3 % ( postop . RCT ) and 3 % ( pre-op . RCT ) ',
 'RCT ) and 3 % ( pre-op . RCT ) from postoperative bleeding , and 6 % ( postop ',
 'postoperative bleeding , and 6 % ( postop . RCT ) and 4 % ( pre-op . RCT ) ',
 'RCT ) and 4 % ( pre-op . RCT ) from delayed wound healing . The patient accrual ',
 'the 6 mg ibandronate group . I.v . ibandronate treatment leads to significant improvements in quality of life , ']

In [217]:
find_wrong_lines(dev_check, dev)

[]

#### Glaucoma


In [208]:
glaucoma_check = check('ecai2020-transformer_based_am/data/glaucoma_test/test.conll')
glaucoma_check

[114, 18541, 24453, 31537]

In [209]:
glaucoma_incorrect_lines = find_wrong_lines(glaucoma_check, 'ecai2020-transformer_based_am/data/glaucoma_test/test.conll')
glaucoma_incorrect_lines

104
18531
24443
31527


['9.8 and 30.6 ± 9.9 mmHg , respectively . Postoperatively , mean IOPs were 11.4 ± 4.9 and 13.6 ± ',
 'least as effective as pilocarpine 2 % t.i.d . in reducing IOP when added to eyes currently on monotherapy with ',
 'the trabeculectomy group ( P = 0.7 ) . Mean postoperative IOP was 13.7+/-2.2 mmHg at 3 months , 14.8+/-3.3 ',
 ', 24.7 mm Hg in the medical group . Mean IOP ( both eyes ) at last follow-up was 18.2 ']

#### Mixed

In [223]:
mx_conll[4].value_counts()

O            16015
I-Premise    11767
I-Claim       3942
B-Premise      388
B-Claim        212
Name: 4, dtype: int64

In [224]:
mixed_check = check('ecai2020-transformer_based_am/data/mixed_test/test.conll')
mixed_check

[32920]

In [225]:
print(find_wrong_lines(mixed_check, mixed))


32910
['glucose levels < 54 mg/dL by 65 % . ( 3.1 vs. 8.9 min , P < 0.001 ) . ']


***( Majority of the problematic lines are in the neoplasm train and test data )***

<a id='comp_clf'></a>
## Argument components classification results

<!-- ### Results from authors

<img src='pics/result_authors.png'>
<img src='pics/results_fine-tune.png'>
<img src='pics/results_all.png'> -->

###   Results: pre-problem identification

||Neoplasm||||Glaucoma||||Mixed|||
|-|-|-|-|-|-|-|-|-|-|-|-|
||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1|
|SciBert+GRU+CRF|.84|.77|.93||.84|.82|.92||.82|.77|.91|
|BioBert+GRU+CRF|.81|.74|.91||.84|.77|.94||.82|.78|.92|
|Bert+GRU+CRF|.82|.72|.91||.83|.80|.89||.83|.77|.90|
|||||||||||||
|SciBert+LSTM+CRF|.70|.73|.90||.70|.72|.92||.69|.74|.90|
|||||||||||||
|mBert+LSTM+CRF|.84|.76|.92|||||||||

###   Results: post-problem identification

||Neoplasm||||Glaucoma||||Mixed|||
|-|-|-|-|-|-|-|-|-|-|-|-|
||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1|
|SciBert+GRU+CRF|.84|.76|.92||.86|.84|.92||.84|.77|.92|
|BioBert+GRU+CRF|.85|.76|.94||.88|.85|.92||.87|.81|.93|
|Bert+GRU+CRF|.84|.73|.93||.85|.80|.90||.84|.77|.90|
|mBert+GRU+CRF|.84|.74|.90||.83|.77|.85||.83|.74|.89|
|||||||||||||
|SciBert+LSTM+CRF|.84|.75|.93||.84|.83|.92||.82|.76|.91|

<!-- BioBert+LSTM+CRF|.83|.76|.93||.83|.82|.91||.82|.80|.92|
Bert+LSTM+CRF|.84|.75|.92||.82|.79|.88||.83|.77|.90|
mBert+LSTM+CRF|.83|.75|.89||.83|.76|.88||.83|.73|.90| -->

E-F1 is always higher than C-F1 => harder to identify claims

In [57]:
preds = pd.read_csv('ecai2020-transformer_based_am/output/sequence_tagging_predictions.conll', sep='\t', header=None)
preds.head()

0  1  2
0          b'we'  O  O
1  b'previously'  O  O
2    b'reported'  O  O
3        b'that'  O  O
4   b'treatment'  O  O

<a id='translation_projection'></a>
# Translations and Projections



* DeepL
* Opus
* Mixed


### IAA

In [16]:
import sklearn
from sklearn.metrics import cohen_kappa_score

# ann1 = [5,3,5,3,3,3,3,3,1,3,3,3,3,3,3,3,3,3,3,3,3,1,3,3,3,3,1,3,3]
# ann2 =[3,3,1,3,3,3,3,3,5,3,3,5,3,3,2,3,1=3,3,3,5,3,5,3=5,3,3,3,1=2=4=5,3,3]


ann1 = [5,3,5,3,3,3,3,3,1,3,3,3,3,3,3,3,3,3,3,3,3,1,3,3,3,3,1,3,3]
ann2 = [3,3,1,3,3,3,3,3,5,3,3,5,3,3,2,3,3,3,3,5,3,5,3,3,3,3,1,3,3]
# a = [3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3]
# b = [3,3,3,3,3,3,3,3,3,3,3,1,3,3,3,3,3,3,3,3,3]
print(1 -cohen_kappa_score(ann1, ann2))



0.6789297658862876


In [15]:

def cohen_kappa(ann1, ann2):
    """Computes Cohen kappa for pair-wise annotators.
    :param ann1: annotations provided by first annotator
    :type ann1: list
    :param ann2: annotations provided by second annotator
    :type ann2: list
    :rtype: float
    :return: Cohen kappa statistic
    """
    count = 0
    for an1, an2 in zip(ann1, ann2):
        if an1 == an2:
            count += 1
    A = count / len(ann1)  # observed agreement A (Po)

    uniq = set(ann1 + ann2)
    E = 0  # expected agreement E (Pe)
    for item in uniq:
        cnt1 = ann1.count(item)
        cnt2 = ann2.count(item)
        count = ((cnt1 / len(ann1)) * (cnt2 / len(ann2)))
        E += count

    return round((A - 1/5) / (1 - 1/5), 4)
cohen_kappa(ann1, ann2)

0.6983

### Neoplasm 
   * train - 4405 lines
   * test - 1251 lines
   * dev - 679 lines

### Glaucoma
   * test - 1247 lines

### Mixed
   * test - 1148 lines
   

## Translations

||train|dev|neoplasm|glaucoma|mixed|
|-|-|-|-|-|-|
|**DeepL**|✅|✅|✅|✅|✅|
|**Opus-mt**|✅|✅|✅|✅|✅|


## Projections

||train|dev|neoplasm|glaucoma|mixed|
|-|-|-|-|-|-|
|**DeepL**|
|simalign|✅|✅|✅|✅|✅|
|awesome|✅|✅|✅|✅|✅|
|**Opus**|
|simalign|✅|✅|✅|✅|✅|
|awesome|✅|✅|✅|✅|✅|




#### Post-processing. Compare

Projections using awesome align mostly does not project labels to the first tokens in the sentence that are articles, conjunctions, etc(Tokens like: 'Por lo tanto' => 'therefore'/ 'thus', etc.). when they are supposed to have a tag (B- or I-). 

Post-processing: if the sentence is full argument in the original data => same sentence is full argument in the translated data. Therefore, we can assign the full argument sentences without projection. Below are the numbers of how many sentence were changes during mentioned process (more readable version is in the end of this [section](#post_table)):

In [157]:
def get_difference(projected_path, corrected_path):
    projected = open(projected_path, 'r')
    corrected = open(corrected_path, 'r')
    
    difference_in_tokens = 0
    difference_in_seq = 0
    p_tags = []
    p_seq_tag = []
    c_tags = []
    c_seq_tag = []
    for p, c in zip(projected, corrected):
        if p != c: 
            difference_in_tokens += 1
        if p != '\n':
            p_seq_tag.append(p.split(' ')[1])
        else:
            p_tags.append(p_seq_tag)
            p_seq_tag = []
        if c != '\n':
            c_seq_tag.append(c.split(' ')[1])

        else:
            c_tags.append(c_seq_tag)
            c_seq_tag = []

    p_tags.append(p_seq_tag)
    c_tags.append(c_seq_tag)
    print(len(p_tags))
    for p, c in zip(p_tags, c_tags):
       
        if set(p) != set(c): 
            difference_in_seq += 1
    
    return difference_in_tokens, difference_in_seq

In [158]:
def count_full_seq(data_path):
    f = open(data_path)
    counter = 0
    num_os = 0
    tags = []
    cur_tag = []
    for line in f:
        if line != '\n':
            cur_tag.append(line.split(' ')[1])
        else:
            tags.append(cur_tag)
            if cur_tag[0].startswith('B') and len(set(cur_tag[1:])) == 1:
#                 print(len(cur_tag))
                counter += 1
            if len(set(cur_tag)) == 1 and cur_tag[0].startswith('O'):
                num_os += 1
            cur_tag = []
            
#     print(len(tags))
    print("Number of O's: ",num_os)
    return counter

### Awesome-align

#### Neoplasm

In [174]:
print('Neoplasm train (deepl & awesome): ')
print('Number of sentences that are full component in original data: ', count_full_seq('antidote-projections/data/neoplasm/train.tsv'))
print('Number of sentences that are full component in projected: ', count_full_seq('antidote-projections/data/projections/deepl_awesome/neoplasm/pprocessed_train.tsv'))
get_difference('antidote-projections/data/projections/deepl_awesome/neoplasm/train.tsv', 'antidote-projections/data/projections/deepl_awesome/neoplasm/pprocessed_train.tsv')

Neoplasm train (deepl & awesome): 
Number of O's:  2345
Number of sentences that are full component in original data:  1769
Number of O's:  2345
Number of sentences that are full component in projected:  1752
4405


(1652, 800)

In [175]:
print('Neoplasm test (deepl & awesome): ')
print('Number of sentences that are full component in original: ', count_full_seq('antidote-projections/data/neoplasm/test.tsv'))
print('Number of sentences that are full component in projected: ', count_full_seq('antidote-projections/data/projections/deepl_awesome/neoplasm/pprocessed_test.tsv'))
get_difference('antidote-projections/data/projections/deepl_awesome/neoplasm/test.tsv', 'antidote-projections/data/projections/deepl_awesome/neoplasm/pprocessed_test.tsv')

Neoplasm test (deepl & awesome): 
Number of O's:  630
Number of sentences that are full component in original:  518
Number of O's:  630
Number of sentences that are full component in projected:  518
1252


(505, 242)

In [ ]:
# 1769   518  262

In [176]:
print('Neoplasm dev (deepl & awesome): ')
print('Number of sentences that are full component in original: ', count_full_seq('antidote-projections/data/neoplasm/dev.tsv'))
print('Number of sentences that are full component in projected: ', count_full_seq('antidote-projections/data/projections/deepl_awesome/neoplasm/pprocessed_dev.tsv'))
get_difference('antidote-projections/data/projections/deepl_awesome/neoplasm/dev.tsv', '/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_awesome/neoplasm/pprocessed_dev.tsv')

Neoplasm dev (deepl & awesome): 
Number of O's:  377
Number of sentences that are full component in original:  262
Number of O's:  377
Number of sentences that are full component in projected:  257
680


(195, 95)

#### Glaucoma

In [177]:
print('Glaucoma (deepl & awesome): ') 
print('Number of sentences that are full component in original: ', count_full_seq('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/glaucoma/test.tsv'))
print('Number of sentences that are full component in projected: ', count_full_seq('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_awesome/glaucoma/pprocessed_test.tsv'))
get_difference('antidote-projections/data/projections/deepl_awesome/glaucoma/test.tsv', '/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_awesome/glaucoma/pprocessed_test.tsv')

Glaucoma (deepl & awesome): 
Number of O's:  682
Number of sentences that are full component in original:  498
Number of O's:  682
Number of sentences that are full component in projected:  498
1248


(341, 167)

#### Mixed

In [179]:
print('Mixed (deepl & awesome): ') 
print('Number of sentences that are full component in original: ', count_full_seq('antidote-projections/data/mixed/test.tsv'))
print('Number of sentences that are full component in projected: ', count_full_seq('antidote-projections/data/projections/deepl_awesome/mixed/pprocessed_test.tsv'))
get_difference('antidote-projections/data/projections/deepl_awesome/mixed/test.tsv', 'antidote-projections/data/projections/deepl_awesome/mixed/pprocessed_test.tsv')

Mixed (deepl & awesome): 
Number of O's:  591
Number of sentences that are full component in original:  478
Number of O's:  591
Number of sentences that are full component in projected:  476
1147


(415, 203)

### Simalign

#### Neoplasm

In [171]:
print('Neoplasm train (deepl & simalign): ')
print('Number of sentences that are full component: ', count_full_seq('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/neoplasm/train.tsv'))
get_difference('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/neoplasm/train.tsv', '/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/neoplasm/pprocessed_train.tsv')

Neoplasm train (deepl & simalign): 
Number of O's:  2345
Number of sentences that are full component:  1769
4405


(466, 88)

In [168]:
print('Neoplasm test (deepl & simalign): ')
print('Number of sentences that are full component: ', count_full_seq('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_awesome/neoplasm/pprocessed_test.tsv'))
get_difference('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/neoplasm/test.tsv', '/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/neoplasm/pprocessed_test.tsv')

Neoplasm test (deepl & simalign): 
Number of O's:  630
Number of sentences that are full component:  518
1252


(189, 92)

In [167]:
print('Neoplasm dev (deepl & simalign): ')
print('Number of sentences that are full component: ', count_full_seq('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_awesome/neoplasm/pprocessed_dev.tsv'))
get_difference('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/neoplasm/dev.tsv', '/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/neoplasm/pprocessed_dev.tsv')

Neoplasm dev (deepl & simalign): 
Number of O's:  377
Number of sentences that are full component:  257
680


(66, 11)

#### Glaucoma

In [169]:
print('Glaucoma (deepl & simalign): ')
print('Number of sentences that are full component: ', count_full_seq('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/glaucoma/pprocessed_test_1.tsv'))
get_difference('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/glaucoma/test.tsv', '/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/glaucoma/pprocessed_test_1.tsv')

Glaucoma (deepl & simalign): 
Number of O's:  682
Number of sentences that are full component:  506
1248


(144, 51)

#### Mixed

In [170]:
print('Mixed (deepl & simalign): ')
print('Number of sentences that are full component: ', count_full_seq('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/mixed/pprocessed_test_1.tsv'))
get_difference('/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/mixed/test.tsv', '/Users/anaryegen/Desktop/UPV/Master_Thesis/antidote-projections/data/projections/deepl_simalign/mixed/pprocessed_test_1.tsv')

Mixed (deepl & simalign): 
Number of O's:  591
Number of sentences that are full component:  480
1147


(248, 90)

* Neoplasm train/test/dev: 1769   518  262
* Glaucoma: 498
* Mixed: 478

<a id='post_table'></a>
### Number of changes

"_s" - simalgin

"_a" - awesome-align

|(DeepL)|train_a|train_s|dev_a|dev_s|neoplasm_a|neoplasm_s|glaucoma_a|glaucoma_s|mixed_a|mixed_s|
|-|-|-|-|-|-|-|-|-|-|-|
|overall|4405|4405|680|680|1252|1252|1248|1248|1147|1147|
|# of full O's|2345|2345|377|377|630|630|692|682|591|591|
|# of full component|1752|703|257|257|518|518|498|506|476|480|
|# of auto-corrections|800|88|95|11|242|92|167|51|203|90|
|# of manual-corrections|140|194|20|25|51|26|26|14|47|26|



<a id='zero-shot'></a>
# Zero-shot
### Without post-processing

||Neoplasm||||Glaucoma||||Mixed|||
|-|-|-|-|-|-|-|-|-|-|-|-|
||||||**Simalign**|||||
||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1|
|mBERT+GRU+CRF|
|*DeepL*|-|-|-||.73|.65|.81||.74|.65|.84|
|*Opus*|.76|.70|.86||.74|.67|.81||.75|.69|.85|
||||||**Awesome**|||||
|*DeepL*|-|-|-||.69|.64|.80||.68|.63|.84|
|*Opus*|.70|.68|.85||.70|.66|.80||.69|.68|.84|


### After post-processing

||Neoplasm||||Glaucoma||||Mixed|||
|-|-|-|-|-|-|-|-|-|-|-|-|
||||||**Simalign**|||||
||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1|
|mBERT+GRU+CRF|
|*DeepL*|.79|.69|.87||.76|.70|.81||.78|.69|.85|
|*Opus*|.79|.69|.88||.76|.68|.80||.78|.70|.84|
||||||**Awesome**|||||
|*DeepL*|.78|.69|.87||.76|.69|.81||.77|.69|.84|
|*Opus*|.78|.69|.87||.75|.68|.80||.78|.70|.84|

### Train and evaluate ES/ES & EN/ES

#### mBERT & BETO (train -ES => test - ES)
||Neoplasm||||Glaucoma||||Mixed|||
|-|-|-|-|-|-|-|-|-|-|-|-|
||||||**Simalign**|||||
||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1|
|**mBERT**|
|*DeepL*|.81|.75|.89||.79|.75|.87||.80|.74|.89|
|*Opus*|.79|.72|.88||.78|.75|.86||.80|.74|.89|
|**BETO**|
|*DeepL*|.79|.74|.90||.78|.79|.89||.78|.72|.90|
|*Opus*|.78|.73|.90||.78|.79|.89||.77|.74|.89|
||||||**Awesome**|||||
|**mBERT**|
|*DeepL*|.80|.74|.88||.78|.75|.87||.79|.74|.89|
|*Opus*|.79|.72|.88||.78|.75|.86||.79|.74|.89|
|**BETO**|
|*DeepL*|.78|.74|.90||.77|.78|.89||.77|.72|.89|
|*Opus*|.77|.73|.89||.78|.79|.89||.77|.73|.89|


<!-- #### BETO
||Neoplasm||||Glaucoma||||Mixed|||
|-|-|-|-|-|-|-|-|-|-|-|-|
||||||**Simalign**|||||
||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1|
|BETO+GRU+CRF|
|*DeepL*|.83|.75|.90||.83|.80|.90||.83|.75|.91|
|*Opus*|.80|.74|.88||.79|.75|.84||.81|.76|.88|
||||||**Awesome**|||||
|*DeepL*|.82|.75|.90||.83|.75|.91||.82|.80|.89|
|*Opus*|.81|.74|.90||.83|.82|.89||.82|.77|.90| -->

#### Train - EN + ES => Test - ES

||Neoplasm||||Glaucoma||||Mixed|||
|-|-|-|-|-|-|-|-|-|-|-|-|
||||||**Simalign**|||||
||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1||F1-macro|C-F1|E-F1|
|*DeepL*|84.35|76.64|88.43||84.54|78.67|87.24||83.90|73.46|88.87|
|*Opus*|84.54|76.02|88.89||84.20|78.69|87.31||83.18|72.15|88.98|
||||||**Awesome**|||||
|*DeepL*|84.58|76.77|88.61||84.62|78.71|87.23||84.03|73.99|88.92|
|*Opus*|84.35|75.76|88.89||84.25|78.74|87.26||83.16|72.46|88.84|

#### Train - EN + ES => Test - EN

||Neoplasm||||Glaucoma||||Mixed|||
|-|-|-|-|-|-|-|-|-|-|-|-|
|EN => EN|84.34|74.68|89.88||83.95|77.91|86.79||83.63|75.42|89.65|
|EN+ES => EN|83.51|73.42|89.38||85.31|81.05|86.73||83.63|74.98|89.25|

<!-- eval_f1_macro = 0.8283145954559271       
eval_f1_micro = 0.88376279715426         
f1_claim = 0.7357030134515669               
f1_evidence = 0.8887772887063897  

# g
eval_f1_macro = 0.8432134457273017               
eval_f1_micro = 0.8783164030688784               
f1_claim = 0.8137134198314234                    
f1_evidence = 0.8623455338715157 

#m 
eval_f1_macro = 0.8320493187989695      
eval_f1_micro = 0.8898684125234978      
f1_claim = 0.7610239471511149           
f1_evidence = 0.8929866304101518  -->

<!-- # Train en+es => test es (deepl+simalign)(mBert)
#Neoplasm
eval_f1_macro = 0.8331153841159322           
eval_f1_micro = 0.8842202282523245  
f1_claim = 0.7642670871490084          
f1_evidence = 0.8870742183598043 

# Glaucoma
eval_f1_macro = 0.8432297937148395
eval_f1_micro = 0.891302317763293    
f1_claim = 0.8001632653061226             
f1_evidence = 0.8766085585903499 

# Mixed
eval_f1_macro = 0.8276006934273226
eval_f1_micro = 0.8814192515435649    
f1_claim = 0.7540607939610434       
f1_evidence = 0.8820181813702546  

#opus+awesome
# n
eval_f1_macro = 0.8231949813663691    
eval_f1_micro = 0.8808427086532109       
f1_claim = 0.7492806205429751       
f1_evidence = 0.8865031462694234 
# g
eval_f1_macro = 0.835482877127567
eval_f1_micro = 0.8869774873467982  
f1_claim = 0.8008611410118407        
f1_evidence = 0.8734213410856407   
# m
eval_f1_macro = 0.8157857067189477  
eval_f1_micro = 0.8754521477015825  
f1_claim = 0.7551700208188759         
f1_evidence = 0.8755925365607665   

# opus+simailgn
#n
eval_f1_macro = 0.8315486985987438
eval_f1_micro = 0.8828713138475568    
f1_claim = 0.7549044265593561          
f1_evidence = 0.8894982317666956 
#g
eval_f1_macro = 0.8417131464135337         
eval_f1_micro = 0.8869200815162793        
f1_claim = 0.8036748882635326               
f1_evidence = 0.8730090605627088  
#m
eval_f1_macro = 0.8230364075842536  
eval_f1_micro = 0.8764506405425773    
f1_claim = 0.7568804159445408         
f1_evidence = 0.8769498273820023   

# deepl+awesome
# NEo
eval_f1_macro = 0.8208313036341793        
eval_f1_micro = 0.8824491492495504      
f1_claim = 0.7614884963273235       
f1_evidence = 0.8844663349751232 
#g
eval_f1_macro = 0.8364950775840218  
eval_f1_micro = 0.8898828978577965           
f1_claim = 0.7977133523887301                
f1_evidence = 0.8748910508586369
#m
eval_f1_macro = 0.8167979303797827
eval_f1_micro = 0.8805727062763969  
f1_claim = 0.7524015694763901      
f1_evidence = 0.8808520079875749 

# Train en+es => test es (deepl+simalign)(BETO)

# deepl+awesome
# neoplasm
eval_f1_macro = 0.817383732726738  
eval_f1_micro = 0.884087804330333  
f1_claim = 0.7551413881748072                
f1_evidence = 0.8873463114754099 

# glaucoma
eval_f1_macro = 0.8216812826486678  
eval_f1_micro = 0.8950146739337577         
f1_claim = 0.804414261460102         
f1_evidence = 0.8817333014124971

#mixed
eval_f1_macro = 0.807591296933762 
eval_f1_micro = 0.8812696145678688        
f1_claim = 0.7207846410684474             
f1_evidence = 0.8867386180374642   

# opus+awesome
# neoplasm
eval_f1_macro = 0.8099494109863706       
eval_f1_micro = 0.8818545133151443         
f1_claim = 0.7401346179226606             
f1_evidence = 0.8889993719907892 

# glaucoma
eval_f1_macro = 0.8176784586783586    
eval_f1_micro = 0.8951697584879243           
f1_claim = 0.7991386735572782               
f1_evidence = 0.88249885605568 

#mixed
eval_f1_macro = 0.8072936890406357 
eval_f1_micro = 0.8798016161770901        
f1_claim = 0.7395671240628094                     
f1_evidence = 0.8830523513753328  -->

<a id='rel_clf'></a>
# Relation classification

### Train

In [4]:
data = pd.read_csv('/ecai2020-transformer_based_am/data/neoplasm/train_relations.tsv', sep='\t', header=None)
data.head()

0                                                  1  \
0  __label__noRel  The overall remission rate was 87% with 31% co...   
1  __label__noRel  The overall remission rate was 87% with 31% co...   
2  __label__noRel  The overall remission rate was 87% with 31% co...   
3  __label__noRel  The overall remission rate was 87% with 31% co...   
4  __label__noRel  The overall remission rate was 87% with 31% co...   

                                                   2  
0  The median survival of all 406 eligible patien...  
1  The overall remission rate, the rate of comple...  
2  In limited disease the estimated percentages o...  
3  Patients with extensive disease survived signi...  
4  In the latter patients the received dose inten...

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14286 entries, 0 to 14285
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       14286 non-null  object
 1   1       14286 non-null  object
 2   2       14286 non-null  object
dtypes: object(3)
memory usage: 335.0+ KB


In [6]:
data[0].value_counts()

__label__noRel      12892
__label__Support     1194
__label__Attack       200
Name: 0, dtype: int64

In [22]:
data[1].nunique()

2266

In [13]:
count = 0
for line in data[1]:
    for char in line:
#     print(line)
        count += 1

print(f'Number of characters in training data: {count}')
    

Number of characters in training data: 1882891


### Test

In [14]:
test = pd.read_csv('/ecai2020-transformer_based_am/data/neoplasm/test_relations.tsv', sep='\t', header=None)
test.head()

0                                                  1  \
0  __label__noRel  Responses (> or = 50% improvement) were seen i...   
1  __label__noRel  Responses (> or = 50% improvement) were seen i...   
2  __label__noRel  Responses (> or = 50% improvement) were seen i...   
3  __label__noRel  Responses (> or = 50% improvement) were seen i...   
4  __label__noRel  Responses (> or = 50% improvement) were seen i...   

                                                   2  
0  The rate of progressive disease was 47%, 21%, ...  
1  Eight (73%) of 11 patients crossing over from ...  
2  The median duration of response from start of ...  
3  The following drug-related adverse effects wer...  
4  No cases of drug-related neutropenic fever, se...

In [27]:
test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4380 entries, 0 to 4379
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       4380 non-null   object
 1   1       4380 non-null   object
 2   2       4380 non-null   object
dtypes: object(3)
memory usage: 102.8+ KB


In [26]:
test[0].value_counts()

__label__noRel      3961
__label__Support     359
__label__Attack       60
Name: 0, dtype: int64

In [23]:
test[1].nunique()

686

In [25]:
count = 0
for line in test[1]:
    for char in line:
#     print(line)
        count += 1

print(f'Number of characters in test data: {count}')
    

Number of characters in test data: 577319


### Dev

In [28]:
dev = pd.read_csv('/Users/anaryegen/Desktop/UPV/servers/ecai2020-transformer_based_am/data/neoplasm/dev_relations.tsv', sep='\t', header=None)
dev.head()

0                                                  1  \
0    __label__noRel  Intervention increased SF36 score by 9.5 point...   
1    __label__noRel  Intervention increased SF36 score by 9.5 point...   
2  __label__Support  Intervention increased SF36 score by 9.5 point...   
3    __label__noRel  Intervention increased SF36 score by 9.5 point...   
4    __label__noRel  Effect size (ES) was 0.63 [0.37; 0.90], 0.29 [...   

                                                   2  
0  Effect size (ES) was 0.63 [0.37; 0.90], 0.29 [...  
1  Anxiety score was shortly minored by intervent...  
2  This 2-week group intervention seemed to durab...  
3  Differences, smaller at 12 months than at six,...  
4  Intervention increased SF36 score by 9.5 point...

In [29]:
dev.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2030 entries, 0 to 2029
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       2030 non-null   object
 1   1       2030 non-null   object
 2   2       2030 non-null   object
dtypes: object(3)
memory usage: 47.7+ KB


In [30]:
dev[0].value_counts()

__label__noRel      1815
__label__Support     185
__label__Attack       30
Name: 0, dtype: int64

In [32]:
dev[1].nunique()

326

### Results

|| precision   | recall | f1-score | support| 
|-| - | - | - | - |   
|**sciBERT**|
||    0.95   |   0.96  |    0.96    |   1815|                                             
||    0.61   |   0.61   |   0.61    |   185 |                                               
||    0.78   |   0.23   |   0.36    |    30 |                                                
|accuracy|    ||                        0.92   |   2030  |                                               
|macro avg|       0.78  |     0.60  |    0.64  |   2030  |                                               
|weighted avg|       0.92  |    0.92   |   0.92   |   2030  |                                               
|**bioBERT**|
||   0.95  |   0.97  |   0.96   |   1815
||   0.68  |   0.64  |   0.66   |    185
||   0.00  |   0.00  |   0.00   |     30
|    accuracy|   ||                         0.93  |  2030|
|   macro avg|       0.54   |   0.54  |    0.54    |  2030|
|weighted avg|       0.91   |   0.93   |  0.92    |  2030|
|**BERT**|
||0.94   |   0.96   |   0.95   |   1815
||0.59   |   0.57   |   0.58   |    185
||0.00   |   0.00   |   0.00   |     30
|    accuracy|  ||                         0.91   |   2030
|   macro avg|       0.51 |     0.51   |   0.51   |   2030
|weighted avg|       0.90  |    0.91   |   0.90   |   2030


In [60]:
#sciBert
eval_f1_macro = 0.6411988822634034         
eval_f1_macro_filtered = 0.7823111439079256

eval_f1_micro = 0.9187192118226601       
eval_f1_micro_filtered = 0.9241482218353644

#bioBert
eval_f1_macro = 0.5397247640802364
eval_f1_macro_filtered = 0.8095871461203545

eval_f1_micro = 0.9261083743842364
eval_f1_micro_filtered = 0.9332340531149169

#Bert
eval_f1_macro = 0.5100361117076702
eval_f1_macro_filtered = 0.7650541675615052

eval_f1_micro = 0.9103448275862069
eval_f1_micro_filtered = 0.9171215880893302

|| precision   | recall | f1-score | support| 
|-| - | - | - | - |   
|**es => es (BETO)**|
||      0.93|      0.98|      0.95|      1814|
||      0.62|     0.42|      0.50|       185|
||      0.00|      0.00|      0.00|        30|
|    accuracy|||                           0.91|      2029|
|   macro avg|       0.52|      0.47|      0.49|      2029|
|weighted avg|       0.89|      0.91|      0.90|      2029|
|**en => es(mBERT)**|
||        0.93   |   0.97  |    0.95  |    1814|
||        0.57   |   0.39  |    0.46  |     185|
||        0.45   |   0.17  |    0.24  |      30|
|    accuracy|  ||                         0.90|      2029|
|   macro avg|       0.65  |    0.51  |    0.55|      2029|
|weighted avg|       0.89  |    0.90  |    0.89|      2029|
   
        

In [ ]:
# es => es
eval_f1_macro = 0.5623686841904604
eval_f1_macro_filtered = 0.7153478980805623
eval_f1_micro = 0.905864958107442
eval_f1_micro_filtered = 0.9121672057725804

#en => es
eval_f1_macro = 0.551885577785158
eval_f1_macro_filtered = 0.705877147165542

eval_f1_micro = 0.9043863972400197
eval_f1_micro_filtered = 0.9111277072442121


In [ ]:
                 precision    recall  f1-score   support                   
                                                                           
           0       0.96      0.95      0.96      3961                      
           1       0.56      0.65      0.60       359                      
           2       0.55      0.55      0.55        60                      
                                                                           
    accuracy                           0.92      4380                      
   macro avg       0.69      0.72      0.70      4380                      
weighted avg       0.92      0.92      0.92      4380                      
                                                                           
eval_f1_macro = 0.7031163805308805
eval_f1_macro_filtered = 0.7796745707963209                                               
eval_f1_micro = 0.9194063926940639                                                        
eval_f1_micro_filtered = 0.924537037037037  

||    | Neoplasm | Glaucoma | Mixed| 
|-| - | - | - | - |   
|SciBERT|      70.3|      0.98|      0.95|      1814|

### Glaucoma

In [5]:
# glaucoma_rel = pd.read_csv('/ecai2020-transformer_based_am/data/glaucoma_test/train_relations.tsv', sep='\t')
glaucoma_rel = pd.read_csv('../argument_mining/data/glaucoma_test/test_relations.tsv', sep='\t', header=None)
glaucoma_rel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3332 entries, 0 to 3331
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       3332 non-null   object
 1   1       3332 non-null   object
 2   2       3332 non-null   object
dtypes: object(3)
memory usage: 78.2+ KB


In [6]:
glaucoma_rel.head()

0                                                  1  \
0    __label__noRel  El porcentaje de reducción de la PIO o la magn...   
1    __label__noRel  El porcentaje de reducción de la PIO o la magn...   
2    __label__noRel  El porcentaje de reducción de la PIO o la magn...   
3    __label__noRel  El porcentaje de reducción de la PIO o la magn...   
4  __label__Support  El porcentaje de reducción de la PIO o la magn...   

                                                   2  
0  En el campo visual, la tasa estimada de cambio...  
1  La tasa estimada de cambio en MD no mostró dif...  
2  No se observaron cambios en la topografía de l...  
3  No hubo pacientes que abandonaran el tratamien...  
4  Tanto el tratamiento con latanoprost como el t...

In [7]:
glaucoma_rel[0].value_counts()

__label__noRel      2986
__label__Support     317
__label__Attack       29
Name: 0, dtype: int64

### Mixed

In [8]:
mixed_rel = pd.read_csv('../argument_mining/data/mixed_test/test_relations.tsv', sep='\t', header=None)
mixed_rel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3332 entries, 0 to 3331
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       3332 non-null   object
 1   1       3332 non-null   object
 2   2       3332 non-null   object
dtypes: object(3)
memory usage: 78.2+ KB


In [9]:
mixed_rel.head()

0                                                  1  \
0    __label__noRel  En los grupos de sildenafilo frente a placebo,...   
1    __label__noRel  En los grupos de sildenafilo frente a placebo,...   
2  __label__Support  En los grupos de sildenafilo frente a placebo,...   
3    __label__noRel  Los cambios en la clase funcional de la OMS y ...   
4    __label__noRel  Los cambios en la clase funcional de la OMS y ...   

                                                   2  
0  Los cambios en la clase funcional de la OMS y ...  
1  El dolor de cabeza, la diarrea y el rubor fuer...  
2  Sildenafilo, además del tratamiento con bosent...  
3  En los grupos de sildenafilo frente a placebo,...  
4  El dolor de cabeza, la diarrea y el rubor fuer...

In [10]:
mixed_rel[0].value_counts()

__label__noRel      3012
__label__Support     296
__label__Attack       24
Name: 0, dtype: int64